src/domain/user.js

```diff
@@ -358,15 +361,16 @@ async function _findByUnique(key, value) {
     }
   })
 
-  function flatten(user) {
-    if (!user.profile) {
-      return user
-    }
-    const { profile, ...rest } = user
-    return { ...rest, ...profile }
-  }
   if (foundUser) {
     return flatten(foundUser)
   }
   return null
 }
+
+function flatten(user) {
+  if (!user.profile) {
+    return user
+  }
+  const { profile, ...rest } = user
+  return { ...profile, ...rest }
+}
```

Comment:
What is `rest`?

Reply:
The rest is everything except the profile object. First, we separate the profile object, and then we merge it with the rest.